src/price_data.py

```diff
@@ -267,33 +268,49 @@ def _get_price_bitpanda_pro(
         # if there were no trades in the requested time frame, the
         # returned data will be empty
         for t in timeframes:
-            end = utc_time
-            begin = utc_time - datetime.timedelta(minutes=t)
+            num_offsets = 0
+            for num_offsets in range(6):
+                # if no trades can be found, move 30 min window to the past
+                window_offset = num_offsets * 30
+                end = utc_time.astimezone(datetime.timezone.utc) \
+                    - datetime.timedelta(minutes=window_offset)
+                begin = end - datetime.timedelta(minutes=t)
+                if num_offsets:
+                    log.warning(
+                        f"No price data found for {base_asset} / {quote_asset} "
+                        f"at {end}, moving {timeframes[-1]} minute window to the past."
+                    )
+
+                # https://github.com/python/mypy/issues/3176
+                params: dict[str, Union[int, str]] = {
+                    "unit": "MINUTES",
+                    "period": t,
+                    # convert ISO 8601 format to RFC3339 timestamp
+                    "from": begin.isoformat().replace('+00:00', 'Z'),
+                    "to": end.isoformat().replace('+00:00', 'Z'),
+                }
+                log.debug(
+                    f"Calling Bitpanda API for {base_asset} / {quote_asset} "
+                    f"price for {t} minute timeframe ending at {end}"
```

Comment:
Add information about `window_offset` in debugging message when `window_offset > 0`.